src/lib/dnssd/Resolver.h

```diff
@@ -423,7 +441,21 @@ class Resolver
     virtual CHIP_ERROR DiscoverCommissioners(DiscoveryFilter filter = DiscoveryFilter()) = 0;
 
     /**
-     * Stop discovery (of commissionable or commissioner nodes).
+     * @brief Start Browising for operational devices
+     *
+     * This will start periodical brwosing of operational devices. The when a new node is
+     * found the delegate set with SetOperationalBrowseDelegate is called. The mDNS queryies
+     * will continue unitil StopDiscovery is called.
```

Comment:
How about interaction with other browse (commissionable/commissioner)? It seems to me like StopDiscovery is a stop for 2 separate things. Lifetime of delegates and interactions with commissionable/commissioner delegates seems to become awkward.

I find the API for the browses generally not ideal: each browse should be independent and have its own delegate that can be acted upon. Something like:

```cpp
/// Here BrowseParams should have:
///   - Type of browse
///   - delegate to call, lifetime of that delegate  tied to the browse handle
CHIP_ERROR StartBrowse(const OperationalBrowseParams &params, BrowseHandle &out_handle);
CHIP_ERROR StartBrowse(const CommissionableBrowseParams &params, BrowseHandle &out_handle);
CHIP_ERROR StopBrowse(BrowseHandle &handle);
```

Building upon the existing global delegates and global stop seems not ideal.

Reply:
I would potentially even add a "timeout" to browse parameters depending on use case. Browsing forever seems wasteful (makes all nodes report their address periodically).

Reply:
> Browsing forever seems wasteful

Browsing forever is exactly what's wanted for https://github.com/project-chip/connectedhomeip/issues/25091.

Reply:
The issue we have is that minmdns has no "known answers" support (because limited memory) so a continous browse will actually force every device to send all its info every time, creating a lot of traffic (that most devices may not hardware-filter out since it is of the "right type").

For MinMDNS we would however be capable to get the packets of "startup announcement" and report new devices without an active browse, so I wonder about the actual use case because maybe registering a "give me unsolicited packet data without initiating a browse" is a good idea ... although I am unclear how that would work on platformdns (probably it would not).

Generally I am somewhat concerned about any "browse forever".